learn/shorts/intro-gh-actions.qmd

```diff
@@ -0,0 +1,369 @@
+    -   Knowing what a Git repository means
+    -   Navigating to a Git repository
+    -   Creating a new file ("adding")
+    -   Saving files ("committing")
+    -   Making edits to files ("committing")
+
+## Necessary software
+
+-   No software is needed.
+-   Need to have a GitHub account.
+
+## Learning goal
+
+> At the end of this, you will be able to describe what an Action is and
+> some basic uses of it, as well as being able to create a simple
+> Action.
+
+## Take home messages
+
+-   GitHub Actions are files with a list of instructions that the GitHub
+    server uses to run to complete a set of tasks.
+-   GitHub Actions are YAML files that must be placed in the
+    `.github/workflows/` folder in a repository.
+
+## Lesson content
+
+### What is a GitHub Action?
+
+At it's core, a [GitHub Action](https://github.com/features/actions) is
+a set of instructions given to a server to run and complete a specific
+sequence of tasks. Before we can continue though, we need to answer the
+question: What is a server?
+
+A server is just like the computer you are using right now to read this
+lesson in your web browser, except that a server is a connected set of
+thousands of computers usually kept somewhere in a big,
+tightly-controlled and managed warehouse. Servers are very powerful,
+large "computers" that is used to run complex and intensive
+computations, and most often are used to run tasks for websites. All of
+the Internet is found on servers across the world.
+
+<!-- TODO: Include a diagram or something here? -->
+
+In the case of GitHub Actions, the server is GitHub's server, and the
+instructions for the tasks to do are kept in a specific file. In order
+to GitHub to know about and use this file, it has to be kept in the
+`.github/workflows/` folder of the repository and the file must be a
+[YAML](https://en.wikipedia.org/wiki/YAML) file (ending in `.yml`):
+
+```         
+reponame/
+└── .github/
+    └── workflows/
+        ├── action-name-1.yml
+        └── action-name-2.yml
+```
+
+In the folder setup above, there are two Actions that GitHub will run:
+`action-name-1` and `action-name-2`.
+
+### General structure of a GitHub Action
+
+GitHub Actions are generally structured and organized into distinct
+sections:
+
+1.  The trigger (`on`) section
+2.  The permissions (`permissions`) section
+3.  The jobs (`jobs`) section
+
+There is also a starting `name` field to tell GitHub what the name or
+title of the Action is, that is usually at the very top of the file.
+
+#### `on` section
+
+The `on` section tells GitHub when the Action should run. For instance,
+to only run the Action when an Issue (like bug reports, questions to
+ask, or tasks to do) is created (called "opened"), you write:
+
+``` yaml
+on:
+  issues:
+    type: opened
+```
+
+::: callout-tip
+The way that commands are written as a shortened uses the `.` to
+separate each level. So for the one above, it would be `on.issues.type`.
```

Comment:
```suggestion
The way that commands are written in official documentation often uses a shortened form. So that a `.` is used to
separate each level. In the example above, in documentation it would be referred to as `on.issues.type`.
```